### PR TITLE
bump(ftp): commons-net 3.11.1 (was 3.8.0)

### PR DESCRIPTION
**Pull request by ennru**
[CVE-2021-37533](https://www.cve.org/CVERecord?id=CVE-2021-37533): Apache Commons Net's FTP client trusts the host from PASV response by default

* https://commons.apache.org/proper/commons-net/security.html
* https://github.com/hierynomus/sshj#release-history

Tests in our `FtpsWithProxyStageTest` started failing which makes https://github.com/apache/commons-net/pull/90 look suspicious. We had to disable the test.

Refs
- "Unsupported or unrecognized SSL message" for FTPS via proxy https://issues.apache.org/jira/browse/NET-718

### CODE REVIEW

**Review by efgpinto — APPROVED** [review]
LGTM

**Comment by ennru** [comment]
Doesn't look too good `javax.net.ssl.SSLException: Unsupported or unrecognized SSL message`.

**Comment by ennru** [comment]
This error is not directly connected to the change of default in 3.9.0, it fails locally with

```
Start of log messages of test that failed with assertion failed: fishForMessage(OnNext(_) or OnComplete) found unexpected message OnError(javax.net.ssl.SSLException: Unsupported or unrecognized SSL message
	at java.base/sun.security.ssl.SSLSocketInputRecord.handleUnknownRecord(SSLSocketInputRecord.java:451)
	at java.base/sun.security.ssl.SSLSocketInputRecord.decode(SSLSocketInputRecord.java:175)
	at java.base/sun.security.ssl.SSLTransport.decode(SSLTransport.java:111)
	at java.base/sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1505)
	at java.base/sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1420)
	at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:455)
	at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:426)
	at org.apache.commons.net.ftp.FTPSClient._openDataConnection_(FTPSClient.java:278)
	at org.apache.commons.net.ftp.FTPClient._retrieveFileStream(FTPClient.java:915)
	at org.apache.commons.net.ftp.FTPClient.retrieveFileStream(FTPClient.java:2841)
	at akka.stream.alpakka.ftp.impl.CommonFtpOperations.$anonfun$retrieveFileInputStream$1(CommonFtpOperations.scala:71)
	at scala.util.Try$.apply(Try.scala:210)```

**Comment by johanandren** [comment]
As in, it was already failing?

**Comment by ennru** [comment]
> As in, it was already failing?

No, as in fails only after the upgrade, but switching back the default the CVE reported doesn't help.

**Comment by ennru** [comment]
There were not many changes in commons-net between 3.8.0 and 3.9.0:
https://issues.apache.org/jira/browse/NET-642?jql=project%20%3D%20NET%20AND%20fixVersion%20%3D%203.9.0

Tests in our `FtpsWithProxyStageTest` fail which makes https://github.com/apache/commons-net/pull/90 look suspicious.

**Comment by sebastian-alfers** [comment]
When trying to reproduce locally, I do get another error:

```
--> [akka.stream.alpakka.ftp.FtpsWithProxyStageTest: listFiles] Start of log messages of test that failed with assertion failed: expected class akka.stream.testkit.TestSubscriber$OnNext, found class akka.stream.testkit.TestSubscriber$OnError (OnError(java.net.ConnectException: Connection refused (Connection refused)
	at java.base/java.net.PlainSocketImpl.socketConnect(Native Method)
```

It is the same on `master` so I assume the setup is broken on my end. @johanandren are you able to reproduce the error from this build? Locally, it is just:
```
./scripts/ftp-servers.sh
```
and then 
```
sbt "ftp/testOnly *.FtpsWithProxyStageTest"
```

**Comment by johanandren** [comment]
Waiting for upstream issue https://issues.apache.org/jira/browse/NET-718

**Comment by ennru** [comment]
Bumped commons-net to 3.10.0 but according to their issue tracker the problem is not fixed.

**Comment by ennru** [comment]
When getting back to this, notice even
- https://github.com/akka/alpakka/pull/3069

**Comment by ennru** [comment]
sshj already bumped with
- https://github.com/akka/alpakka/pull/3258

File: project/Dependencies.scala [hunk 1]
```diff
@@ -197,7 +197,7 @@ object Dependencies {
 
   val Ftp = Seq(
     libraryDependencies ++= Seq(
-        "commons-net" % "commons-net" % "3.8.0", // ApacheV2
+        "commons-net" % "commons-net" % "3.10.0",
```
Comment (ennru) [inline project/Dependencies.scala:200]:
```suggestion
        "commons-net" % "commons-net" % "3.11.1",
```

File: ftp/src/test/java/akka/stream/alpakka/ftp/FtpsWithProxyStageTest.java [hunk 2]
```diff
@@ -21,7 +21,7 @@
 import java.util.concurrent.CompletionStage;
 import java.util.function.Function;
 
-public class FtpsWithProxyStageTest extends BaseFtpSupport implements CommonFtpStageTest {
+@Ignore public class FtpsWithProxyStageTest extends BaseFtpSupport implements CommonFtpStageTest {
```
Comment (johanandren) [inline ftp/src/test/java/akka/stream/alpakka/ftp/FtpsWithProxyStageTest.java:24]: Would be good to add a comment explaining why it is ignored

**Review by johanandren — APPROVED** [review]
LGTM

**Pull request merged**